### ISSUE TITLE
Fix rounding when encoding timestamps

### ISSUE THREAD

**Issue by tailhook** [edgedb/edgedb-net#27]
According to edgedb/edgedb#4885 conversion should round up to nearest even.

This should happen when encoding .Net native time format into a millisecond precision:
```
1970-01-01T00:00:00.0000005 -> 1970-01-01T00:00:00.000000
1970-01-01T00:00:00.0000015 -> 1970-01-01T00:00:00.000002
```

Also check `LocalTime`, `Duration` and `LocalDatetime` types.

This should happen across all the range of the values, see [Rust tests](https://github.com/edgedb/edgedb-rust/blob/master/edgedb-protocol/tests/datetime_chrono.rs) for more comprehensive set of tests (but adjust precision conversion and time format accordingly). And [the docs PR](https://github.com/edgedb/edgedb/pull/4881) for more info.

**Comment by quinchs** [comment]
Merged in #28, I'll close when a release comes out with the changes

**Comment by quinchs** [comment]
Published in 1.0.5
